code/modules/mob/living/carbon/combat.dm

```diff
@@ -163,6 +163,8 @@
 			isTackling = FALSE	//Safety from throw_at being a jerk
 		else
 			tackleGetHurt()
+			if(airlockEmagTackled(src,Obstacle))
+				Obstacle.emag_act()
```

Comment:
This is not at all how object-oriented code works.
You want to define `/atom/proc/tackled(mob/living/user)`, and have its implementation do nothing.
Then, override it for airlocks, like so:
`/obj/machinery/door/airlock/tackled(mob/living/user)`
In that override, you would implement the code that checks whether `user` is wearing an emag, and call `emag_act` accordingly.

Reply:
Thanks for the clarification - I'll work on this tomorrow

Reply:
Better attempt at this in latest commit